src/Utility/Filter.php

```diff
@@ -0,0 +1,126 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         5.0.1
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Utility;
+
+use JsonException;
+use Stringable;
+
+/**
+ * Methods for converting mixed values to specific data types, ensuring a type-safe approach to data manipulation.
+ * This utility is useful for safely narrowing down the data types.
+ */
+class Filter
```

Comment:
Do we need a class for these functions? PHP has namespaced functions now, we could use those instead of having a class.

Reply:
I am not sure. https://github.com/cakephp/cakephp/pull/17295/commits/a2f668425f977276bdd66eef6c75dca579e80603 ?